src/payment/strategies/bolt/bolt-payment-strategy.ts

```diff
@@ -1,29 +1,131 @@
+            const state = this._store.getState();
+            const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+            if (!paymentMethod || !paymentMethod.initializationData.publishableKey) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            const { publishableKey } = paymentMethod.initializationData;
+
+            this._boltClient = await this._boltScriptLoader.load(publishableKey, paymentMethod.config.testMode);
+        }
+
         return Promise.resolve(this._store.getState());
     }
 
     deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._boltClient = undefined;
+
         return Promise.resolve(this._store.getState());
     }
 
-    async execute(payload: OrderRequestBody, _options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (this._useBoltClient) {
+            return this._executeWithBoltClient(payload, options);
+        } else {
+            return this._executeFullCheckoutTakeOver(payload, options);
+        }
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    private async _executeWithBoltClient(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+
+        if (useStoreCredit !== undefined) {
+            await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        }
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(payment.methodId, options));
+        const paymentMethod = state.paymentMethods.getPaymentMethod(payment.methodId);
+
+        if (!paymentMethod || !paymentMethod.clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const orderToken = paymentMethod.clientToken;
+
+        const transaction: BoltTransaction = await new Promise((resolve, reject) => {
+            const onSuccess = (transaction: BoltTransaction,  callback: () => void) => {
+                resolve(transaction);
+                callback();
+            };
+
+            const onClose = () => {
+                reject(new PaymentMethodCancelledError());
+            };
+
+            const callbacks = {
+                success: onSuccess,
+                close: onClose,
+            };
+
+            if (!this._boltClient) {
```

Comment:
can this happen at the beginning of the method `_executeWithBoltClient`? same way we validate the payment